lib/fluent/plugin/out_s3.rb

```diff
@@ -54,9 +54,8 @@ def configure(conf)
       end
       @compressor.configure(conf)
 
-      # TODO: use Plugin.new_formatter instead of TextFormatter.create
-      conf['format'] = @format
-      @formatter = TextFormatter.create(conf)
+      @formatter = Plugin.new_formatter(@format)
```

Comment:
`Plugin.new_formatter` is available since v0.10.58 so should update gem dependency.